docs/batch_request.md

```diff
@@ -40,4 +40,7 @@ meRequestItem, _ := batchCollection.AddBatchRequestStep(*meRequest)
 eventsRequestItem, _ := batchCollection.AddBatchRequestStep(*eventsRequest)
 
 batchResponse, _ := batchCollection.Send(context.Background(), client.GetAdapter())
+
+// print the first response
+fmt.Println(GetBatchResponseById[User](batchResponse, "1", CreateUserFromDiscriminatorValue)) // returns a serialized response
```

Comment:
do you think we could get the factory method by reflection + conventions so we don't need to ask it to the user now that we have the generic type parameter?

Reply:
We would not be able to add the factory method since there is no way to access methods that are not on a type. To use this pattern.
We can however have a global variable as a registry and pass the discriminator function from the sdk i.e

```go

var ParsableFactoryRegistry : constructor absser.ParsableFactory  // Add this on implementing SDK

func GetBatchResponseById[T serialization.Parsable](resp BatchResponse, itemId string) (T, error) {
	var res T
	if ParsableFactoryRegistry == nil {
		return res , errors.New("ParsableFactoryRegistry is nil ")
	}
----
```

Reply:
If we have type information in the node, we can use a single factory e.g this implementation `[models.CreateEntityFromDiscriminatorValue](https://github.com/microsoftgraph/msgraph-sdk-go/blob/c2137cc334f18950034177de1a7c8f414f79dd25/models/entity.go#L22)` .

Reply:
Agreed. In this case it might not be possible to use reflection to get the factory method

Reply:
thanks for getting to the bottom of this, resolving